index.js

```diff
@@ -156,15 +156,22 @@ RedirectableRequest.prototype.setTimeout = function (msecs, callback) {
     this.on("timeout", callback);
   }
 
+  function setSocketTimeout(socket) {
```

Comment:
This function has to be defined locally here as opposed to globally just to have access to the  `msecs` argument passed by the client code.
Moreover the logic to set the socket timeout is refactored into its own method to be able to serve redirect requests which don't need the other `startTimer()` logic.